R/utilities-goodness-of-fit.R

```diff
@@ -738,32 +736,58 @@ plotResidualsQQPlot <- function(data,
   return(qqPlot)
 }
 
-getSimulationTimeRanges <- function(simulation, path, timeUnit) {
-  firstApplicationRange <- list(name = "firstApplicationRange", values = NULL)
-  lastApplicationRange <- list(name = "lastApplicationRange", values = NULL)
-  applicationTimes <- 0
-
+getSimulationTimeRanges <- function(simulation, path, timeUnit, settings, logFolder) {
+  # Initialize output
+  timeRanges <- list(total = list(name = ApplicationRanges$total,
+                                  keep = isTRUE(settings$applicationRanges[[ApplicationRanges$total]]),
+                                  values = NULL),
+                     firstApplication = list(name = ApplicationRanges$firstApplication,
+                                             keep = FALSE,
+                                             values = NULL),
+                     lastApplication = list(name = ApplicationRanges$lastApplication,
+                                            keep = FALSE,
+                                            values = NULL))

```

Comment:
@pchelle  I don't unerstand why `keep = FALSE` for the first/last AppRange. Can you explain please?
As **per default**, all 3 ranges should be activated (same as before)

Reply:
Sorry, I should have commented this.
The initialization here defines the default for single administration, for which total = firstApplication = lastApplication.
The chunk from lines 783-787 overwrites this in the case of multiple administrations.

In each case, the default was already defined in `settings` which is a `TaskSettings` object.
When initializing `settings` for plot time profile task, `settings$applicationRanges$total`, `settings$applicationRanges$firstApplication`, and `settings$applicationRanges$lastApplication` are all TRUE.

Reply:
ok, thx for the clarification!